tools/docker/22.04/Dockerfile

```diff
@@ -0,0 +1,184 @@
+                       gdb \
+                       cmake \
+                       clang \
+                       clang-format \
+                       dpdk \
+                       dpdk-dev \
+                       libdpdk-dev \
+                       libhdf5-dev \
+                       libhdf5-serial-dev \
+                       libfftw3-dev \
+                       libairspy-dev \
+                       libboost-test-dev \
+                       libevent-dev \
+                       libnuma-dev \
+                       libssl-dev \
+                       wget \
+                       flex \
+                       bison \
+                       iwyu \
+                       && \
+    apt-get clean && apt-get autoclean
+
+
+# Install kotekan python dependencies
+RUN python3 -m pip install --upgrade pip && \
+    python3 -m pip install --upgrade distro-info && \
+    python3 -m pip install --upgrade --force-reinstall setuptools && \
+    python3 -m pip install --upgrade wheel && \
+    python3 -m pip install --no-cache-dir numpy && \
+    python3 -m pip install --no-cache-dir pkgconfig && \
+    python3 -m pip install --no-cache-dir --upgrade cython && \
+    python3 -m pip install --no-cache-dir click==8.0.2 && \
+    python3 -m pip install --no-cache-dir black==19.10b0 && \
+    python3 -m pip install --no-cache-dir cmake_format
+
+RUN python3 -m pip install h5py \
+                           hdf5plugin \
+                           bitshuffle
+RUN python3 -m pip install msgpack \
+                              future \
+                              requests \
+                              pyyaml \
+                              tabulate \
+                              pytest \
+                              pytest-xdist \
+                              pytest-cpp \
+                              pytest-localserver \
+                              pytest-timeout \
+                              posix_ipc
+RUN python3 -m pip install --ignore-installed flask
+
+
+# Install comet, mysql, redis for comet tests
+RUN apt-get update && \
+    apt-get install -y mysql-client \
+        libmysqlclient-dev && \
+    apt-get clean && apt-get autoclean
+RUN python3 -m pip install git+https://github.com/chime-experiment/comet.git
+RUN apt-get update && \
+    apt-get install -y redis && \
+    apt-get clean && apt-get autoclean
+
+
+# CUDA / GPU tools
+RUN wget https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64/cuda-keyring_1.0-1_all.deb
+RUN dpkg -i cuda-keyring_1.0-1_all.deb
+RUN apt -y update
+RUN apt install -y --no-install-recommends cuda && apt clean
+
+ENV PATH /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/cuda/bin
+
+RUN git clone https://github.com/kmsmith137/gputils.git \
+    && cd gputils \
+    && make install
+
+RUN git clone https://github.com/kmsmith137/n2k.git \
+  && cd n2k \
+  && make install
+
+RUN apt install -y --no-install-recommends \
```

Comment:
Maybe just add a comment we should change this to ROCm if possible.

Reply:
My understanding is that the `rocm/dev-ubuntu-22.04` image is only available with Rocm versions >= 5.3, which don't work with the CHIME code.
https://hub.docker.com/r/rocm/dev-ubuntu-22.04/tags

So we need to use the older `rocm/dev-ubuntu-18.04` image, which supports much older Rocm versions (eg 3.7).

Reply:
Thanks -- going to leave this as-is in this PR.